### DISCUSSION TITLE
How do I upgrade to V19

### DISCUSSION THREAD

**Discussion by chesterflaps** [awawa-dev/HyperHDR#430] [General]
Hi all.

Can someone point me in the right direction for upgrading from V18 to V19 please? I cannot seem to find a tutorial and there are loads of files in the download section and not sure which to get and what to do with them.

**Comment by awawa-dev** [comment]
Hi
You have everything explained here on this page: https://github.com/awawa-dev/HyperHDR/wiki/Installation#Linux

**Reply by chesterflaps** [reply]
Thanks. So it's basically the same as a fresh install? Will it carry over my settings and do I have to fully format my SD card again to get it to work?

**Reply by awawa-dev** [reply]
you can avoid  formatting your sd card: ssh to your rpi, remove old version and install new version (adjust link in the manual  from v18 to v19beta1) the settings will be preserved.

**Reply by chesterflaps** [reply]
Thank you but I do not understand any of that. I have Putty which is SSH but it doesn't allow me to see individual files. Is there a file or folder called 'manual'? Not sure what you mean by adjust link in the manual from v18 to v19beta1

**Reply by chesterflaps** [reply]
Sounds easy to say 'remove old version and install new one but how do I actually do that?

**Reply by awawa-dev** [reply]
Can't help further with that subject. Please respect our time. Screenshot from wiki:
![obraz](https://user-images.githubusercontent.com/69086569/204843453-80b16517-e619-4867-a210-c3c594208d96.png)

**Reply by vkodsi** [reply]
Hi, I have just finished my (Linux) update script : interested ?

**Comment by vkodsi** [comment]
Here is my **_HyperHDR_update.sh_** script :

**==========================================================**
available here : [https://github.com/awawa-dev/HyperHDR/wiki/LinuxUpdateScript](url)
**==========================================================**

**Reply by vkodsi** [reply]
I have just updated my post to make it compliant with other architectures than raspberry ;)

**Reply by awawa-dev** [reply]
@vkodsi I haven't time to test it yet but it looks promising 👍

**Reply by vkodsi** [reply]
Let me know if you have an issue since I have only tested it on 2 Raspberrys (a Pi 4 and a Pi zero 2 W so both arm v7...)

**Reply by vkodsi** [reply]
By the way I just saw that you wrote 

> For Rpi Zero 2 W choose aarch64 instead!

...but it works on RPi 2 W with arm v7 (that's what I have tested on...)

**Reply by vkodsi** [reply]
So I'm gonna update my script in order to take your perf optimization propositions (aarch64 vs arm v7) into account using 
> lshw -short | awk '/system/{print $2, $3, $4}'

in order to find wath hardware  the script is running on ;)

**Reply by awawa-dev** [reply]
When you install HyperHDR from inside of running OS, you don't have an option to choose between aarch64 and armv7: theoretically you are able to run armv7 code on most aarch64 systems (if not disabled in the configuration) but you will run into troubles with dependencies: different naming/versions/architecture of depended libraries that are not come with HyperHDR.

**Reply by chesterflaps** [reply]
Do I just copy and paste that whole script into Putty and it will do it all?

**Reply by chesterflaps** [reply]
I've removed the previous version using the Wiki instructions but the next step to download keeps saying not found

wget https://github.com/awawa-dev/HyperHDR/releases/download/HyperHDR-19.0.0.0beta1-Linux-armv7l.deb

That is what I am putting into Putty

**Reply by awawa-dev** [reply]
https://github.com/awawa-dev/HyperHDR/releases/download/v19.0.0.0beta1/HyperHDR-19.0.0.0beta1-Linux-armv7l.deb

**Reply by chesterflaps** [reply]
Thank you. Done that and all was going well until the attached came up in Putty. Does this matter?
![IMG_3043](https://user-images.githubusercontent.com/86977015/205093135-503f38ca-c5f5-465b-8702-10c7180926ee.JPG)

**Reply by chesterflaps** [reply]
ignore the remove part at the bottom

**Reply by awawa-dev** [reply]
you are installing armv7 version on your aarch64 system: that wont work as I've just explained 
use this: https://github.com/awawa-dev/HyperHDR/releases/download/v19.0.0.0beta1/HyperHDR-19.0.0.0beta1-Linux-aarch64.deb

**Reply by chesterflaps** [reply]
Got it downloaded and none of the settings carried over. I have set them all up again because I luckily had  a note of them.

The issue I have now is the LEDs are not lighting up 
![Screenshot 2022-12-01 155618](https://user-images.githubusercontent.com/86977015/205099866-3306c69f-a786-45d3-98ba-cc57add762a8.jpg)

**Reply by chesterflaps** [reply]
It's totally failed now and I cannot log onto the dashboard as it fails to load.

Now allowed me onto dashboard but LEDs not lighting up even though the dashboard shows them lit up in the visulaliser

**Reply by awawa-dev** [reply]
Excusez moi...you haven't finished your installation yet and so before you make any further assumption I can assure you: your settings are safe. Since your LED worked before then you must run it as ```root``` user, but now since it doesn't work and you don't have your old settings then it's probably run as 'pi' user now (and it cant use ```root``` settings). Read the FAQ on the main page because I officially don't support such configuration where you drive ws281x led strip directly from the Rpi.

**Reply by chesterflaps** [reply]
Where can I get help with this then? I thought this was a forum where people help other people. I really appreciate your help and your software and do not expect you to spend hours answering my questions, there are other people on this forum I am asking for help from.

**Reply by awawa-dev** [reply]
![obraz](https://user-images.githubusercontent.com/69086569/205105553-5d1413a1-513f-4cfc-8757-530bd2107f18.png)

**Reply by chesterflaps** [reply]
I'm not blaming you for anything, I am asking for help. Thank you

**Reply by chesterflaps** [reply]
Dare I ask, where is the FAQ? I've searched and cannot find it

**Reply by chesterflaps** [reply]
Think I found it and got it back on. thank you for your help, it really did help me getting it working again.

I know I ask alot of questions but if I don't know something I will always try and work it out for myself but when it comes to coding, ssh etc, it is a complete minefield when you don't know what you're doing. I am learning as I go.

thanks again.

Now to look into the whole LUTs thing.

**Reply by awawa-dev** [reply]
First: it is clearly stated in the project status and in many places here that the solution you are using is not recommended and not supported by this project and yet...
secondly: despite this, I try to help in such situations, you have been shown the place where the solution is, but you ignored it and made claims to the whole world
third: because this unsupported solution was working for you before the update, so you can't say that you are just learning new things. You must have set it up in the past so are we to expect that you will come back to us with the same problem in a while?

Regardless, I'm glad you finally got your system up and running after the update 👍

**Reply by chesterflaps** [reply]
First: Please do not get defensive. When I say I have a problem, it is not be saying the software is bad, it's because I am looking for assistance.
Second: A forum is a place for people to come together and share knowledge and ask questions without being made to feel stupid. Everybody has a different level of knowledge. If I have a question which is answered on here and then someone else comes across the same problem, they can read the post and hopefully resolve the issue without having to create a new post.
Third: When I first built my setup, I followed a youtube video to the letter so basically just copied what someone else did. Like following a satnav. You know where you're going but you are being guided the whole way. So I am learning as I go.

Thanks again for your help and your software but may I suggest you don't get so defensive in future.

**Reply by vkodsi** [reply]
Keep cool guys, we're only talking about fun stuff here, right ? ;P

**Reply by vkodsi** [reply]
> Do I just copy and paste that whole script into Putty and it will do it all?

Nope, you have to put the content into a file (you're free to choose whatever name you like...), then give this file execution rights (using 'chmod a+x _your_script_name_') and then run it and enjoy ;)

**Comment by awawa-dev** [comment]
@vkodsi Could you paste and share your script in the wiki? https://github.com/awawa-dev/HyperHDR/wiki/LinuxUpdateScript 
I'm afraid it will be hard to find it in this thread.

**Reply by vkodsi** [reply]
Done!

**Reply by awawa-dev** [reply]
Thank you!
